desc/objectives/linear_objectives.py

```diff
@@ -291,8 +303,11 @@ def build(self, eq, use_jit=True, verbose=1):
         # use given targets and weights if specified
         if self.target.size == modes.shape[0] and None not in self.target:
             self.target = self._target[modes_idx]
-        if self.weight.size == modes.shape[0] and self.weight != np.array(1):
+        if self.weight.size == modes.shape[0] and self.weight.size > 1:
```

Comment:
I feel like we can get rid of these conditionals. We can assume that the user is either passing in targets and weights of the correct size, or target=None. I think another source of confusion is that in the parent class we explicitly broadcast target and weight to dim_f when I'm pretty sure we don't need to (and it would make it less confusing if we didn't)

The only thing we'd need to check is 
```
if target is None:
    target = eq.Rb_lmn[modes]
```